internal/cloudapi/v2/v2_test.go

```diff
@@ -947,6 +947,42 @@ func TestComposeCustomizations(t *testing.T) {
 	}`, "id")
 }
 
+func TestComposeRhcSubscription(t *testing.T) {
+	srv, _, _, cancel := newV2Server(t, t.TempDir(), []string{""}, false, false)
+	defer cancel()
+
+	test.TestRoute(t, srv.Handler("/api/image-builder-composer/v2"), false, "POST", "/api/image-builder-composer/v2/compose", fmt.Sprintf(`
+	{
+		"distribution": "%s",
+		"customizations": {
+			"subscription": {
+				"organization": "2040324",
+				"activation_key": "my-secret-key",
+				"server_url": "subscription.rhsm.redhat.com",
+				"base_url": "http://cdn.redhat.com/",
+				"insights": false,
+				"rhc": true
```

Comment:
I think this case should result in a 400. As rhc will automatically subscribe you to insights.

Reply:
The way I was thinking of it is that rhc overrides any insights selection. If it returns an error, and requires the front end to set it to true that implies that the frontend has control over it. I also documented it that way in the openapi yaml just so it would be clear.

I can change it if you're sure, but I think it is fine as it is.

Reply:
Ah sorry, I should've read the description on the endpoint. Then it's fine.